corehq/apps/users/models.py

```diff
@@ -1216,6 +1214,7 @@ def clear_quickcache_for_user(self):
 
         self.get_by_username.clear(self.__class__, self.username)
         self.get_by_user_id.clear(self.__class__, self.user_id)
+        username_to_user_id.clear(self.username)
```

Comment:
like the `.clear`? no, but i don't think we have to worry about that too much. the cache returns incorrectly when a user deletes a mobile worker and then creates a new one with the same username